Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -17,28 +17,43 @@ public struct DatabaseSpecification: Codable {
 }
 
 /// Internal intermediate result of a ListDatabases command.
-internal enum ListDatabasesResults {
+public enum ListDatabasesResults {
     /// Includes the names and sizes.
     case specs([DatabaseSpecification])
 
     /// Only includes the names.
     case names([String])
 }
 
+/// Options for "listDatabases" operations
+public struct ListDatabasesOptions: Encodable {
+    /// Specifies whether to only return databases for which the user has privileges,
+    /// if false MongoDB will return an error for unauthorized users.
+    public let authorizedDatabases: Bool
+
+    public init(_ authorizedDatabases: Bool = true) {
```

Comment:
I don't think we need to provide the default value of true here. generally, we try to avoid hard-coding any server default values into drivers. instead, when the user specifies no value, we send no value, and allow the server to handle what the default behavior should be.
this makes us more future-proofed against any change in the server default. also, server versions before this option was introduced might error if we send it when it isn't supported, so we should avoid doing so without the user explicitly choosing to send it.

instead, let's make the struct's `authorizedDatabases` property an optional, and this initializer can be written as
```swift
public init(authorizedDatabases: Bool? = nil) {
    self.authorizedDatabases = authorizedDatabases
}
```

similarly to my other comment, we want to require the parameter label here as well. if we add more options to this struct in the future the labels will make it clearer which parameter corresponds to which option.

Reply:
I think I can lean on the default init so I don't need to make one anymore, or is it good practice to have one now incase its needed for different behavior in the future?

Reply:
unfortunately I think the default generated initializer has access level `internal` and we need to write one ourselves to make it public.
you can test this though by trying to create an instance of the struct from somewhere in the tests that does not `@testable` import. (we might want to add some tests for using this option anyway)

Reply:
this is kind of annoying to test behaviorally due to the nature of testing auth related stuff, but we could at least do some assertions in the style of those you wrote for `allowDiskUse` to see if we're correctly sending it in the commands to the server

Reply:
Yea I was just on the hunt for a nice `try self.withAuthTestNamespace` method 😅 maybe that's a good future helper!

Reply:
Is the synthesized one `internal` even if all the fields are `public`? I was really hoping we'd be able to remove them all eventually; it would be disappointing if that never can happen. :cry:

Reply:
I just tested looks like that is the case, `internal` until declared otherwise.